token.go

```diff
@@ -164,36 +166,125 @@ type memoryStore struct {
 	validUntil   time.Time
 }
 
-func (s *memoryStore) AccessToken() string {
+// persistentStore implements tokenStore backed by disk.
+type persistentStore struct {
+	mu           sync.RWMutex // protects the following members
+	AccessToken  string
+	RefreshToken string
+	ValidUntil   time.Time
+}
+
+func (s *memoryStore) GetAccessToken() (string, error) {
 	s.mu.RLock()
 	defer s.mu.RUnlock()
-	return s.accessToken
+	return s.accessToken, nil
 }
 
-func (s *memoryStore) RefreshToken() string {
+func (s *persistentStore) GetAccessToken() (string, error) {
```

Comment:
Please group the functions for `persistentStore` together near the type declaration.